org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.java

```diff
@@ -41,13 +42,13 @@
 /**
  * @author Sven Efftinge - Initial contribution and API
  */
-public class CodeActionService implements ICodeActionService2 {
+public class CodeActionService extends QuickFixCodeActionService {
 	@Inject
 	private IChangeSerializer serializer;
 
 	@Override
 	public List<Either<Command, CodeAction>> getCodeActions(ICodeActionService2.Options options) {
-		List<Either<Command, CodeAction>> actions = new ArrayList<>();
+		List<Either<Command, CodeAction>> actions = super.getCodeActions(options);
```

Comment:
shouldnt the to upper case which mimics the quickfix be dropped here?

Reply:
ahh no, that one is a command

Reply:
What do you mean? This accumulates the quickfix and the hard-coded code-action in `org.eclipse.xtext.ide.tests.testlanguage.ide.server.CodeActionService.fixInvalidName(Diagnostic, Options)`.
This leads to two actions being returned for the same issue ID. One with the type QuickFix from the super implementation. What would you expect?

Reply:
both things do a to upper case.
do we need both.
if i read your quickfixcode action service correct it does a code acction not a command?

Reply:
so maybe we should change it to

command.setTitle("Make '" + string + "' upper case (Command)");

Reply:
but the impl is bogus as there is no edit command in
TestLanguageExecutableCommandService

Reply:
=> have no idea why the current impl is as it is

Reply:
Ok, I'll just drop the "old" one and use the quickfix action instead in the tests. Then we have a real implementation test.

Reply:
i wonder if we want to have a testcase for command code action anyway thus having both would be better.
even if its not completely implemented